scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,1038 @@
+        for (Iterator<TimedOutRequest> iterator = waitUntilQueue.iterator(); iterator.hasNext();) {
+            TimedOutRequest waitRequest = iterator.next();
+            waitRequest.substractTime(timeSpentWaiting);
+            logger.trace("Updated timeouted request: " + waitRequest);
+            if (testWaitFunction(service, waitRequest.getRequest())) {
+                // predicate is met, serve the request
+                service.serve(waitRequest.getRequest());
+                iterator.remove();
+            } else if (waitRequest.getRemainingTimeout() <= 0) {
+                // predicate is not met and timeout expired, serve the request with a timeout exception
+                logger.trace("Timeout observed: " + waitRequest.getRequest().getMethodName());
+                service.serveWithException(waitRequest.getRequest(),
+                                           new TimeoutException("Timeout of " + waitRequest.getConfiguredTimeout() +
+                                                                " ms expired while waiting for predicate"));
+                iterator.remove();
+            }
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> T evaluateClosure(String closureDefinition, Class<T> type) throws CompilationException {
+        try {
+            return (T) shell.evaluate(closureDefinition + " as " + type.getCanonicalName());
+        } catch (CompilationFailedException e) {
+            // CompilationFailedException contains instances which are not serializable
+            throw new CompilationException(StackTraceUtil.getStackTrace(e));
+        }
+    }
+
+    /**
+     * Execute the waitUntil function corresponding to the provided request
+     *
+     * If an error occurs, serve the request to unblock the caller with the exception attached
+     *
+     * @param service active object service
+     * @param request request to process
+     * @return true if the result of the waitUntil call is true or if an exception occurred
+     */
+    private boolean testWaitFunction(Service service, Request request) {
+        boolean dontNeedWaiting;
+        try {
+
+            dontNeedWaiting = executeWaitPredicateFunction(request);
+        } catch (Exception e) {
+            service.serveWithException(request, e);
+            // in case an exception occurs, don't delay the request, execute it immediately to notify the client
+            dontNeedWaiting = true;
+        }
+        return dontNeedWaiting;
+
+    }
+
+    /**
+     * Extract from the request the parameters corresponding to the waitUntil method call, then execute waitUntil with these parameters
+     * @param request request used to extract parameters
+     * @return the result of the waitUntil method call
+     */
+    private boolean executeWaitPredicateFunction(Request request) throws InvocationTargetException,
+            MethodCallExecutionFailedException, InvalidChannelException, CompilationException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)